src/components/UI/TopMenu.tsx

```diff
@@ -1,19 +1,117 @@
-import { Breadcrumbs, Link } from "@mui/material";
+import { AppBar, Box, Button, Toolbar, Typography } from "@mui/material";
+import { useState } from "react";
 
-const TopMenu: () => JSX.Element = () => {
-  return (
-    <Breadcrumbs aria-label="breadcrumb">
-      <Link underline="hover" color="primary" href="/">
-        All Todos
-      </Link>
-      <Link underline="hover" color="secondary" href="/new-todo">
-        Add Todo
-      </Link>
-      <Link underline="hover" color="secondary" href="/edit-todo">
-        Edit Todo
-      </Link>
-    </Breadcrumbs>
-  );
+import { checkTokenValidity } from "../features/tokenValidityCheck";
+import SignIn from "../Authentication/SignIn";
+import SignUp from "../Authentication/SignUp";
+import ThemeSwitcher from "./ThemeSwitcher";
+import UserProfile from "../Authentication/UserProfile";
+
+const TopMenu = ({ token }) => {
+  const [openSignInModal, setOpenSignInModal] = useState(false);
+  const [openSignUpModal, setOpenSignUpModal] = useState(false);
+  const [openUserModal, setOpenUserModal] = useState(false);
+
+  if (checkTokenValidity(token)) {
```

Comment:
Here would be better to have 2 separate components. One for auth and other non auth